### PR TITLE
Synchronize keycloak from auth service's DB upon login

### PR DESCRIPTION
**Pull request by sbose78**
fixes https://github.com/fabric8-services/fabric8-auth/issues/250

These changes should be tested in the `fabric8` realm, because the `update user in keycloak` happens with a PAT ( and not the user token ) so that the `EmailVerified` attribute can be updated.

**Scenario 1** - User is not present in DB ( prod-preview )
1. Drop the local auth db.
2. Assuming you are pre-approved, login using `http://localhost:8089/api/login?redirect=http://localhost:8089/api/status`
3. Verify from `http://localhost:8089/api/users?filter[username]=YOUR_USERNAME` that the values are same as those in keycloak. Ensure `EmailVerified` is same in both places.

**Scenario 2** - Assuming user is already present in DB ( production )

4. Go to keycloak , open the user's profile and modify his `email address` and `Bio`
5. Login using `http://localhost:8089/api/login?redirect=http://localhost:8089/api/status`
6. Verify in Keycloak the changed values have been overwritten by those present in the auth db.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-265-1`

File: account/name_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	fullName = "john doe mike  brown"
+	firstName, lastName = account.SplitFullName(fullName)
+	assert.Equal(t, "john", firstName)
+	assert.Equal(t, "doe mike  brown", lastName)
+
+	fullName = "john, doe mike  brown"
+	firstName, lastName = account.SplitFullName(fullName)
+	assert.Equal(t, "john,", firstName)
+	assert.Equal(t, "doe mike  brown", lastName)
```
Comment (alexeykazakov) [inline account/name_test.go]: also test `account.SplitFullName(""`) and `account.SplitFullName("John")`

File: login/service.go [hunk 2]
```diff
@@ -785,6 +895,67 @@ func generateGravatarURL(email string) (string, error) {
...
+}
+
+// equalsKeycloakUserProfileAttributes verifies the response from keycloak's user profile
+// and returns true if it matches with the user information managed locally by the auth service.
+func (keycloak *KeycloakOAuthProvider) equalsKeycloakUserProfileAttributes(ctx context.Context, accessToken string, identity account.Identity, userAPIEndpoint string) (bool, error) {
+	profileEqual := true
+
+	retrievedUserProfile, err := keycloak.keycloakProfileService.Get(ctx, accessToken, userAPIEndpoint)
+	if err != nil {
+		log.Error(nil, map[string]interface{}{
```
Comment (alexeykazakov) [inline login/service.go]: log.Error(ctx,...)

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-265-8`

**Pull request merged**